tests/integration/suite/test_workloads.py

```diff
@@ -310,6 +312,39 @@ def test_statefulset_workload_volumemount_subpath(admin_pc):
         assert e.value.error.status == 422
 
 
+def test_workload_redeploy(admin_pc, remove_resource):
+    client = admin_pc.client
+    ns = admin_pc.cluster.client.create_namespace(
+        name=random_str(),
+        projectId=admin_pc.project.id)
+    remove_resource(ns)
+    name = random_str()
+    workload = client.create_workload(
+        name=name,
+        namespaceId=ns.id,
+        scale=1,
+        containers=[{
+            'name': 'one',
+            'image': 'nginx',
+        }],
+        annotations={
+            "cattle.io/timestamp":
+                datetime.datetime.now().isoformat()}, )
+    remove_resource(workload)
+
+    def workload_active():
+        workloads = client.list_workload(uuid=workload.uuid).data
+        return workloads[0].state == 'active'
+
+    wait_for(workload_active,
```

Comment:
I think we can already observe that the default time in wait_for (45 seconds) is not long enough. See these two breaks: https://drone-pr.rancher.io/rancher/rancher/6875, https://drone-publish.rancher.io/rancher/rancher/2202